# The Wilcoxon signed-rank test

- Perform a Mann-Whitney U test.
- Perform a Wilcoxon signed-rank test.
- Interpret the output of these tests.

A non-parametric statitsical test employed to compare paired samples to determine whether the population mean ranks differ.

Consider the following scenario: CU Medical Consultants tested the effectiveness of a new medical treatment for anxiety. The levels of anxiety of seven patients were measured before and after treatment in a pilot study. The results, out of 30, are presented in the following table:

|Before|After|
|---|---|
22|24|
20|21|
25|27|
30|26|
22|23|
24|25|
26|25|

The hypotheses are:
- $H_0$: The median difference between anxiety levels before and after treatment is zero (the treatment has no effect).
- $H_a$: The median difference between anxiety levels before and after treatment is not zero (the treatment has an effect).

In [ ]:
# Import the necessary libraries.
import numpy as np
from scipy.stats import wilcoxon

In [ ]:
# Example data for Wilcoxon signed-rank test
# The two sets of measurements are related/ paired (e.g., before and after a treatment)
data_before = np.array([22, 20, 25,
                        30, 22, 24, 26])
data_after = np.array([24, 21, 27, 26,
                       23, 25, 25])

In [ ]:
# Perform Wilcoxon signed-rank test.
wilcoxon_stat, wilcoxon_p = wilcoxon(data_before,
                                     data_after)

# View the output.
print("Wilcoxon statistic:", wilcoxon_stat)
print("Wilcoxon p-value:", wilcoxon_p)

Wilcoxon statistic: 9.5
Wilcoxon p-value: 0.46875


The output indicates a test statistic value is 9.5. The $p$-value is approximately 0.4688. This $p$-value is also greater than 0.05, suggesting that there is no significant difference between the 'before' and 'after' conditions of the paired data based on the provided data.

# Key information
The interpretation of the $p$-values would depend on the $\alpha$ level you set for your test (commonly 0.05 for a 5% significance level).​